offchain/dispatcher/src/sender.rs

```diff
@@ -0,0 +1,220 @@
+            INITIAL_BACKOFF,
+        ));
+        let signer_layer =
+            SignerMiddleware::new(retry_layer, conditional_signer);
+        Ok(Arc::new(signer_layer))
+    }
+
+    /// Creates the tx-manager instance.
+    /// NOTE: does not try to reinstantiate the tx-manager more than once.
+    async fn tx_manager(
+        conditional_signer: &ConditionalSigner,
+        provider_url: String,
+        database_path: String,
+        chain: Chain,
+    ) -> Result<TxManager> {
+        let middleware =
+            Self::middleware(conditional_signer.clone(), provider_url)?;
+        let result = tx_manager!(new, middleware, database_path, chain);
+        let tx_manager =
+            if let Err(TransactionManagerError::NonceTooLow { .. }) = result {
+                warn!("Nonce too low! Clearing the tx-manager database.");
+                tx_manager!(force_new, middleware, database_path, chain)?
+            } else {
+                let (tx_manager, receipt) = result?;
+                trace!("Database claim transaction confirmed: `{:?}`", receipt);
+                tx_manager
+            };
+        Ok(tx_manager)
+    }
+}
+
+// Public.
+impl ClaimSender {
+    pub async fn new(config: &DispatcherConfig) -> Result<Self> {
+        let chain: Chain = (&config.tx_config).into();
+
+        let conditional_signer =
+            ConditionalSigner::new(chain.id, &config.auth_config)
+                .await
+                .expect("Failed to initialize the transaction signer");
+
+        let tx_manager = Self::tx_manager(
+            &conditional_signer,
+            config.tx_config.provider_http_endpoint.clone(),
+            config.tx_config.database_path.clone(),
+            chain,
+        )
+        .await?;
+
+        let authority = {
+            let (provider, _mock) = Provider::mocked();
+            let provider = Arc::new(provider);
+            Authority::new(
+                config.rollups_deployment.authority_address,
+                provider,
+            )
+        };
+
+        Ok(Self {
+            tx_manager,
+            confirmations: config.tx_config.default_confirmations,
+            priority: config.priority,
+            from: conditional_signer.address(),
+            authority,
+        })
+    }
+}
+
+#[async_trait]
+impl Sender for ClaimSender {
+    #[instrument(level = "trace")]
+    async fn submit_claim(
+        self,
+        dapp_address: Address,
+        rollups_claim: RollupsClaim,
+    ) -> Result<Self> {
+        let transaction = {
+            let submittable_claim =
+                SubmittableClaim(dapp_address, rollups_claim);
+            let call = self
+                .authority
+                .submit_claim(submittable_claim.into())
+                .from(self.from);
+            let to = match call.tx.to().expect("tx `to` should not be null") {
+                NameOrAddress::Address(a) => *a,
+                _ => panic!("expected address, found ENS name"),
```

Comment:
I think we should avoid expect and panic here as well.

Reply:
If I'm not mistaken, these are here because otherwise we would have internal errors. So they are basically assertions. @GCdePaula is that right?